python/tests/test_lowlevel.py

```diff
@@ -417,6 +466,37 @@ def yield_tables(self, ts):
         for table in ts.tables.name_map.values():
             yield table.ll_table
 
+    @pytest.mark.parametrize(
+        "table_name",
+        tskit.TABLE_NAMES,
+    )
+    def test_table_extend_bad_args(self, ts_fixture, table_name):
+        table = getattr(ts_fixture.tables, table_name)
+        ll_table = table.ll_table
+        ll_table_copy = table.copy().ll_table
+
+        with pytest.raises(
+            _tskit.LibraryError,
+            match="Tables can only be extended using rows from a different table",
+        ):
+            ll_table.extend(ll_table, row_indexes=[])
+        with pytest.raises(TypeError):
+            ll_table.extend(None, row_indexes=[])
+        ll_table.extend(ll_table_copy, row_indexes=None)
+        with pytest.raises(ValueError):
+            ll_table.extend(ll_table_copy, row_indexes=5)
+        with pytest.raises(_tskit.LibraryError, match="out of bounds"):
+            ll_table.extend(ll_table_copy, row_indexes=[-1])
+        with pytest.raises(_tskit.LibraryError, match="out of bounds"):
+            ll_table.extend(ll_table_copy, row_indexes=[1000])
+        with pytest.raises(_tskit.LibraryError, match="out of bounds"):
+            ll_table.extend(ll_table_copy, row_indexes=range(10000000, 10000001))
+
+        # Uncastable types
+        for dtype in [np.uint32, np.int64, np.uint64, np.float32, np.float64]:
+            with pytest.raises(TypeError, match="Cannot cast"):
+                ll_table.extend(ll_table_copy, row_indexes=np.array([0], dtype=dtype))
```

Comment:
Good to add a few other types of bad array, like ``[[0, 1], [2, 3]]``, ``[None]``, etc.

Reply:
Yeah, good idea!